layers/meta-tegra-support/recipes-devtools/vpi/vpi1-samples_1.0.15.bb

```diff
@@ -0,0 +1,89 @@
+
+DEPENDS = "libnvvpi1 cmake-native opencv"
+
+do_configure() {
+    :
+}
+
+do_compile() {
+    export vpi_DIR="${RECIPE_SYSROOT}/usr/share/vpi1/cmake"
+    export OpenCV_DIR="${RECIPE_SYSROOT}/usr/lib/cmake/opencv4"
+    export CPP="${@cuda_extract_compiler('CXX', d)[0]}"
+    CXX_EXTRA="${@cuda_extract_compiler('CXX', d, prefix='')[1]}"
+    export CUDA_PATH=${STAGING_DIR_NATIVE}/usr/local/cuda-${CUDA_VERSION}
+    PATH=$CUDA_PATH/bin:$PATH
+    export CPPFLAGS="${CXX_EXTRA} ${CXXFLAGS} -I${CUDA_PATH}/include"
+    export LDFLAGS="-L${CUDA_PATH}/lib ${LDFLAGS}"
+
+    oldwd="$PWD"
+    for subdir in ${VPI_SAMPLES}; do
+        cd "$subdir"
+        cmake .
+        oe_runmake
+        cd "$oldwd"
+    done
+}
+
+do_install() {
+    install -d ${D}/${VPI_SAMPLES_PATH}
+    cp -R --preserve=mode,timestamps ${B}/${VPI_SAMPLES_PATH}/assets ${D}/${VPI_SAMPLES_PATH}
+    install -d ${D}/${VPI_SAMPLES_PATH}/bin
+    install -m 0755 ${B}/${VPI_SAMPLES_PATH}/01-convolve_2d/vpi_sample_01_convolve_2d ${D}/${VPI_SAMPLES_PATH}/bin
+    chrpath -d ${D}/${VPI_SAMPLES_PATH}/bin/vpi_sample_01_convolve_2d
+    install -m 0755 ${B}/${VPI_SAMPLES_PATH}/02-stereo_disparity/vpi_sample_02_stereo_disparity ${D}/${VPI_SAMPLES_PATH}/bin
+    chrpath -d ${D}/${VPI_SAMPLES_PATH}/bin/vpi_sample_02_stereo_disparity
+    install -m 0755 ${B}/${VPI_SAMPLES_PATH}/03-harris_corners/vpi_sample_03_harris_corners ${D}/${VPI_SAMPLES_PATH}/bin
+    chrpath -d ${D}/${VPI_SAMPLES_PATH}/bin/vpi_sample_03_harris_corners
+    install -m 0755 ${B}/${VPI_SAMPLES_PATH}/04-rescale/vpi_sample_04_rescale ${D}/${VPI_SAMPLES_PATH}/bin
+    chrpath -d ${D}/${VPI_SAMPLES_PATH}/bin/vpi_sample_04_rescale
+    install -m 0755 ${B}/${VPI_SAMPLES_PATH}/05-timing/vpi_sample_05_timing ${D}/${VPI_SAMPLES_PATH}/bin
+    chrpath -d ${D}/${VPI_SAMPLES_PATH}/bin/vpi_sample_05_timing
+    install -m 0755 ${B}/${VPI_SAMPLES_PATH}/06-klt_tracker/vpi_sample_06_klt_tracker ${D}/${VPI_SAMPLES_PATH}/bin
+    chrpath -d ${D}/${VPI_SAMPLES_PATH}/bin/vpi_sample_06_klt_tracker
+    install -m 0755 ${B}/${VPI_SAMPLES_PATH}/07-fft/vpi_sample_07_fft ${D}/${VPI_SAMPLES_PATH}/bin
+    chrpath -d ${D}/${VPI_SAMPLES_PATH}/bin/vpi_sample_07_fft
+    install -m 0755 ${B}/${VPI_SAMPLES_PATH}/09-tnr/vpi_sample_09_tnr ${D}/${VPI_SAMPLES_PATH}/bin
+    chrpath -d ${D}/${VPI_SAMPLES_PATH}/bin/vpi_sample_09_tnr
+    install -m 0755 ${B}/${VPI_SAMPLES_PATH}/10-perspwarp/vpi_sample_10_perspwarp ${D}/${VPI_SAMPLES_PATH}/bin
+    chrpath -d ${D}/${VPI_SAMPLES_PATH}/bin/vpi_sample_10_perspwarp
+    install -m 0755 ${B}/${VPI_SAMPLES_PATH}/11-fisheye/vpi_sample_11_fisheye ${D}/${VPI_SAMPLES_PATH}/bin
+    chrpath -d ${D}/${VPI_SAMPLES_PATH}/bin/vpi_sample_11_fisheye
+    install -m 0755 ${B}/${VPI_SAMPLES_PATH}/12-optflow_lk/vpi_sample_12_optflow_lk ${D}/${VPI_SAMPLES_PATH}/bin
+    chrpath -d ${D}/${VPI_SAMPLES_PATH}/bin/vpi_sample_12_optflow_lk
+}
+
+RDEPENDS_${PN} += "libnvvpi1 cuda-cudart cuda-cufft tegra-libraries opencv ffmpeg"
```

Comment:
Most of this is unnecessary - bitbake will automatically pick up the shared library dependencies.
The ffmpeg dependency should be optional via PACKAGECONFIG.